### DISCUSSION TITLE
Discussion on using custom parameters in pytest.main()

### DISCUSSION THREAD

**Discussion by Fish-watching-the-sea** [pytest-dev/pytest#11857] [Q&A] [answered]
I have a runtime parameter `--use_https` defined using pytest hooks. It is defined as follows:

./tests/conftest.py:

```python
def pytest_addoption(parser):
    parser.addoption("--use_https", action="store", dest="use_https", default="0", help="use_https: use HTTPS?")
```

I can run tests using the following two methods in CMD.

* `pytest --use_https 0`
* `pytest --use_https=0`

But in `pytest.main()` I can only run the test using `pytest.main(['--use_https=0'])`. When using `pytest.main(['--use_https 0'])` to run the test, an error will be reported:

`ERROR: file or directory not found: --use_https 0`

In daily use, I found that you can use spaces to separate the flag and value in `pytest.main()`. For example I can use `pytest.main(["-m blocker"])` to run all test cases marked with `@pytest.mark.blocker` decorator.

**Comment by RonnyPfannschmidt** [comment, marked as answer]
This is expected behavior

The form without the equal sign must use 2 strings, one for the options name and one for the value


What you did is the equivalent of `pytest "--my-option 0"`

Use `pytest.main(['--use_https', '0'])`
